extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -644,82 +646,88 @@ export class AjaxMonitor extends BaseTelemetryPlugin implements IDependenciesPlu
             }
 
             function _instrumentXhr():void {
-                if (_supportsAjaxMonitoring(_self) && !_disableAjaxTracking && !_xhrInitialized) {
-                    // Instrument open
-                    _hookProto(XMLHttpRequest, "open", {
-                        ns: _evtNamespace,
-                        req: (args:IInstrumentCallDetails, method:string, url:string, async?:boolean) => {
-                            if (!_disableAjaxTracking) {
-                                let xhr = args.inst as XMLHttpRequestInstrumented;
-                                let ajaxData = xhr[strAjaxData];
-                                if (!_isDisabledRequest(xhr, url) && _isMonitoredXhrInstance(xhr, true)) {
-                                    if (!ajaxData || !ajaxData.xhrMonitoringState.openDone) {
-                                        // Only create a single ajaxData (even when multiple AI instances are running)
-                                        _openHandler(xhr, method, url, async);
+                _self._addHook(onConfigChange(_extensionConfig, () => {
+                    _disableAjaxTracking = !!_extensionConfig.disableAjaxTracking;
+                    _enableRequestHeaderTracking = _extensionConfig.enableRequestHeaderTracking;
+
+                    if (_supportsAjaxMonitoring(_self) && !_disableAjaxTracking && !_xhrInitialized) {
```

Comment:
We probably want to put this part `if (_supportsAjaxMonitoring(_self)` outside of the onConfigChange as this won't change as it's based on the support of the runtime.

Reply:
i.e. don't even "hook" if XHR is not supported